src/pepr/config.ts

```diff
@@ -1,4 +1,4 @@
-const isZarfEnv = process.env.UDS_DOMAIN !== "###ZARF_VAR_DOMAIN###";
+const isZarfEnv = process.env.UDS_DOMAIN === "###ZARF_VAR_DOMAIN###";
```

Comment:
I'll have to poke at this some more - maybe we just need a different check. I tossed a debug log in there while doing a UDS deploy to dump `process.env.UDS_DOMAIN` and it was still the zarf template string, not the actual value. I think with pepr dev it was undefined if I remember right, not sure on pepr deploy. Will test those scenarios out.

Reply:
Scenarios and what `process.env.UDS_DOMAIN` is set to:
- `npx pepr dev`: undefined
- `npx pepr deploy`: `###ZARF_VAR_DOMAIN###`
- `uds run test-uds-core`: `uds.dev`
- `UDS_PKG=authservice uds run test-single-package`: `###ZARF_VAR_DOMAIN###`

So the issue is single capability overlap with `npx pepr deploy`. With what's on `main` it will assume istio on single capability, with what's here on my branch it will assume NO istio on `npx pepr deploy` (both of which are the opposite of what we want). I checked `process.env.UDS_WITH_ISTIO` and we'd hit the same overlap with those as well

Not sure if there's fundamentally any difference between `npx pepr build` + `zarf p deploy` (what is used by `uds run test-single-package`) and `npx pepr deploy`? If not we are probably looking at a runtime check against the cluster or (feels janky) we toss a `--set` on https://github.com/defenseunicorns/uds-core/blob/authservice-add/tasks/deploy.yaml#L27 for something like `SINGLE_CAPABILITY=true` that we can check on the pepr config side. Going to mess with this and see if I can figure out a happy path that doesn't seem clunky.

Reply:
This is the check we ended up with - https://github.com/defenseunicorns/uds-core/pull/153/commits/70054945600b966d5593129f7aeb9fb84aa60132

If handles the undefined case correctly + sets domain on the single capability deploy test. Not sure its the cleanest solve, seems a bit hidden, but without checking against the cluster for istio CRDs/pods/ns I'm not sure we have anything else to go off of.